src/coreclr/gc/gc.cpp

```diff
@@ -36315,41 +36311,37 @@ void gc_heap::make_unused_array (uint8_t* x, size_t size, BOOL clearp, BOOL rese
 // Clear memory set by make_unused_array.
 void gc_heap::clear_unused_array (uint8_t* x, size_t size)
 {
-    // Also clear the sync block
-    *(((PTR_PTR)x)-1) = 0;
-
-    ((CObjectHeader*)x)->UnsetFree();
-
-#ifdef HOST_64BIT
+#ifndef HOST_64BIT
+        UNREFERENCED_PARAMETER(size);
+        // Also clear the sync block
+        *(((PTR_PTR)x)-1) = 0;
+		((CObjectHeader*)x)->UnsetFree();
+#else
 
 #if BIGENDIAN
 #error "This won't work on big endian platforms"
 #endif
 
     // The memory could have been cleared in the meantime. We have to mirror the algorithm
     // from make_unused_array since we cannot depend on the object sizes in memory.
-    size_t size_as_object = (uint32_t)(size - free_object_base_size) + free_object_base_size;
-
-    if (size_as_object < size)
-    {
-        uint8_t * tmp = x + size_as_object;
-        size_t remaining_size = size - size_as_object;
+        uint8_t * tmp = x;
+        size_t remaining_size = size;
 
         while (remaining_size > UINT32_MAX)
         {
             size_t current_size = UINT32_MAX - get_alignment_constant (FALSE)
                 - Align (min_obj_size, get_alignment_constant (FALSE));
 
+            *(((PTR_PTR)tmp)-1) = 0;
```

Comment:
Is it really necessary to clear these interim syncblocks? Can we keep "Also clear the syncblock" before `#ifndef HOST_64BIT` and delete it here?